src/Services/MarketTradingDataService.cs

```diff
@@ -19,89 +19,137 @@ public MarketTradingDataService(ICandlesHistoryServiceProvider candlesHistorySer
+        private async
+            Task<(IReadOnlyDictionary<string, CandlesHistoryResponseModel> spotCandles,
+                IReadOnlyDictionary<string, CandlesHistoryResponseModel> mtCandles)> GetLastDayCandlesAsync(int periodInHours = 24)
+        {
+            var spotCandlesTask = GetLastDayCandlesAsync(MarketType.Spot, periodInHours);
+            var mtCandlesTask = GetLastDayCandlesAsync(MarketType.Mt, periodInHours);
+
+            var spotCandlesData = await spotCandlesTask;
+            var mtCandlesData = await mtCandlesTask;
+
+            return (spotCandles: spotCandlesData, mtCandles: mtCandlesData);
+        }
+
+        private async Task<(CandlesHistoryResponseModel spotCandles, CandlesHistoryResponseModel mtCandles)> GetLastDayCandlesAsync(
+            string assetPair, int periodInHours = 24)
+        {
+            var spotCandlesTask = GetLastDayCandlesAsync(MarketType.Spot, assetPair, periodInHours);
+            var mtCandlesTask = GetLastDayCandlesAsync(MarketType.Mt, assetPair, periodInHours);
+
+            var spotCandlesData = await spotCandlesTask;
+            var mtCandlesData = await mtCandlesTask;
+
+            return (spotCandles: spotCandlesData, mtCandles: mtCandlesData);
+        }
+
+        private async Task<IReadOnlyDictionary<string, CandlesHistoryResponseModel>> GetLastDayCandlesAsync(MarketType market, int periodInHours = 24)
         {
             var assetPairs = await GetAvailableAssetPairsAsync(market);
             var candlesService = _candlesHistoryServiceProvider.Get(market);
-
-            // Take current hour to obtain actual last trade price
-            var to = DateTime.UtcNow.AddHours(1); // exclusive
-            var from = to - TimeSpan.FromHours(25); // inclusive
+            var to = DateTime.UtcNow; // exclusive
+            var from = to - TimeSpan.FromHours(periodInHours + 1); // inclusive
 
             var candles = await candlesService.TryGetCandlesHistoryBatchAsync(assetPairs, CandlePriceType.Trades, CandleTimeInterval.Hour, from, to);
 
             return candles ?? new Dictionary<string, CandlesHistoryResponseModel>();
         }
 
-        private async Task<CandlesHistoryResponseModel> GetLastDayCandlesAsync(MarketType market, string assetPair)
+        private async Task<CandlesHistoryResponseModel> GetLastDayCandlesAsync(MarketType market, string assetPair, int periodInHours = 24)
         {
             var assetPairs = await GetAvailableAssetPairsAsync(market);
             var candlesService = _candlesHistoryServiceProvider.Get(market);
-
-            // Take current hour to obtain actual last trade price
-            var to = DateTime.UtcNow.AddHours(1); // exclusive
-            var from = to - TimeSpan.FromHours(25); // inclusive
+            var to = DateTime.UtcNow; // exclusive
+            var from = to - TimeSpan.FromHours(periodInHours + 1); // inclusive
```

Comment:
Should be FromHours(periodInHours), otherwise two candles will be requested, when periodInHours == 1